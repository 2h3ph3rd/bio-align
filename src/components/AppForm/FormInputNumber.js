import * as React from 'react'
import TextField from '@mui/material/TextField'
import FormControl from '@mui/material/FormControl'
import InputAdornment from '@mui/material/InputAdornment'
import LooksOneIcon from '@mui/icons-material/LooksOne'

export default function FormInputNumber(props) {
  const [value, setValue] = React.useState(props.default)

  React.useEffect(() => {
    setValue(props.default)
  }, [props.default])

  const handle = (e) => {
    const value = e.target.value
    setValue(value)
    if (!isError(value) && !isNaN(value)) {
      console.log(value)
      props.handle(parseInt(value))
    }
  }

  const isError = (v) => {
    return (isNaN(parseInt(v)) && v !== '-') || v < -10 || v > 10
  }

  return (
    <FormControl margin="normal" sx={props.style}>
      <TextField
        disabled={props.disabled}
        onChange={handle}
        label={props.label}
        value={value}
        error={isError(value)}
        helperText={isError(value) ? '' : ''}
        variant="outlined"
        data-testid="value-input"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <LooksOneIcon />
            </InputAdornment>
          ),
        }}
      />
    </FormControl>
  )
}
