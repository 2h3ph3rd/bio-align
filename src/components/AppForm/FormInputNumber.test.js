import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import FormInputNumber from './FormInputNumber'

const fakeHandle = () => {}
const fakeLabel = 'test'
const fakeDefault = 'TEST'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <FormInputNumber
        default={fakeDefault}
        label={fakeLabel}
        handle={fakeHandle}
      ></FormInputNumber>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test('default value param', () => {
  render(
    <FormInputNumber
      default={fakeDefault}
      label={fakeLabel}
      handle={fakeHandle}
    ></FormInputNumber>
  )

  const input = screen.getByTestId('value-input').querySelector('input')
  expect(input).toHaveValue(fakeDefault)
})

test('handle change param', () => {
  const newValue = '2'
  render(
    <FormInputNumber
      default={fakeDefault}
      label={fakeLabel}
      handle={fakeHandle}
    ></FormInputNumber>
  )

  const input = screen.getByTestId('value-input').querySelector('input')
  expect(input).toBeInTheDocument()

  fireEvent.change(input, { target: { value: newValue } })
  expect(input).toHaveValue(newValue)
})
