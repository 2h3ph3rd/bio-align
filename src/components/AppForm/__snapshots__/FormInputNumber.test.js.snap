// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-marginNormal css-ufpu9p-MuiFormControl-root"
>
  <div
    className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    data-testid="value-input"
  >
    <label
      className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-error MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
    >
      test
    </label>
    <div
      className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary Mui-error MuiInputBase-formControl MuiInputBase-adornedStart css-1q6at85-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          className="notranslate"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LooksOneIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-5 14h-2V9h-2V7h4v10z"
          />
        </svg>
      </div>
      <input
        aria-invalid={true}
        autoFocus={false}
        className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="TEST"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-1z7n62"
        >
          <span>
            test
          </span>
        </legend>
      </fieldset>
    </div>
    
  </div>
</div>
`;
